Update Trace Agent macrobenchmark

igoragoli opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.

-->
### What does this PR do?

Updates the Trace Agent macrobenchmark. Changes are described in the corresponding [PR](https://github.com/DataDog/benchmarking-platform/pull/31) in the benchmarking-platform repo.

<!--
* A brief description of the change being made with this pull request.
* If the description here cannot be expressed in a succinct form, consider
  opening multiple pull requests instead of a single one.
-->

### Motivation
We are using a new set of tags in the benchmarking platform, and a new tool to execute experiments.

<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes
- CI Pipeline that runs one of the new macrobenchmarks: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/337473219
- Results on a testing dashboard: https://ddstaging.datadoghq.com/dashboard/vu2-f8r-kts/benchmarking-platform--trace-agent?refresh_mode=paused&tpl_var_bp_app%5B0%5D=%2A&tpl_var_bp_project%5B0%5D=%2A&tpl_var_ci_pipeline_id%5B0%5D=20592268&from_ts=1695828796235&to_ts=1695829471000&live=false
- More testing was performed in the context of the benchmarking-platform repo, please see the linked PR.

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

igoragoli committed:
Update Trace agent macrobenchmarking CI config

ddyurchenko commented on .gitlab/benchmarks/macrobenchmarks.yml:
```diff
@@ -13,20 +13,15 @@ variables:
     - ./steps/run-benchmarks.sh
-    - "./steps/upload-results-to-s3.sh || :"
   artifacts:
     name: "reports"
```
Reference: https://github.com/DataDog/dd-trace-rb/blob/master/.gitlab/benchmarks.yml

For artifacts uploading, please use

```
artifacts:
    name: "artifacts"
    when: always
    paths:
      - platform/artifacts/
    expire_in: 3 months
```

ddyurchenko commented on .gitlab/benchmarks/macrobenchmarks.yml:
```diff
@@ -38,48 +33,56 @@ variables:
   variables:
-    TRACE_AGENT_VERSION: main
     TRACE_AGENT_ENDPOINT: v04
     TRACE_AGENT_CPUS: 2
```
Let's discuss this in PR in benchmarking-platform repo, mb it is better to use `TRACE_AGENT_CPUS: 40-41` to simplify scripts.

igoragoli committed:
Adapt `TRACE_AGENT_CPUS` values to bp-runner

As bp-runner, the tool as of now used for running macrobenchmarks,
takes in cpusets such as `0-1` or `30-34`.

igoragoli committed:
Set `when: always` to artifacts; set new name and new path

igoragoli committed:
Remove slash from `/platform` in `git clone`

igoragoli commented:
## Updates
- Resolved conversations (modified job artifacts name, path and set schedule to `always`; modified `git clone` command)
- Resolved issues in the benchmarking-platform repo
 
## Testing
- Can be see in the following PR: https://github.com/DataDog/benchmarking-platform/pull/31

ddyurchenko reviewed (approved):
Much better!

ddyurchenko committed:
Merge branch 'main' into igoragoli/update-trace-agent-macrobenchmark
